src/index.js

```diff
@@ -43,9 +42,15 @@ const DataSelectorFinder = require('./domain/common/DataSelectorFinder');
 
 const BlockListener = require('./infrastructure/blockchain/BlockListener');
 
+const { RequestRepositoryFactory, ResponseRepositoryFactory } = require('./infrastructure/persistance');
+
+if (process.env.PERSISTANCE === 'MONGODB') {
+  mongoose.connect(`mongodb://${process.env.DATABASE_URL}/${process.env.DATABASE_NAME}`, { useNewUrlParser: true });
+}
+
 const logger = new Logger();
-const requestRepository = new RequestRepository();
-const responseRepository = new ResponseRepository();
+const requestRepository = RequestRepositoryFactory.create(process.env.PERSISTANCE, logger);
```

Comment:
i suggest to create line: const { PERSISTANCE } = process.env; // and use it on this 2 lines